### DISCUSSION TITLE
Can I use the “glaze” library without using “struct”?

### DISCUSSION THREAD

**Discussion by meftunca** [stephenberry/glaze#1033] [Q&A] [answered]
I need to use it independently of “struct”

**Comment by stephenberry** [comment, marked as answer]
Certainly, you can use it with the standard template library or similar containers, or you can write [custom serialization](https://github.com/stephenberry/glaze/blob/main/docs/custom-serialization.md), or you can define custom interfaces using `glz::custom` for your class member functions (see README).

Or, if you need completely generic JSON where you know nothing about the data being read, you can use [json_t](https://github.com/stephenberry/glaze/blob/main/docs/generic-json.md)

**Reply by meftunca** [reply]
You are great, obviously I want to use the beve library when I develop my database project. Of course I need the json library to break down the requests :).

**Reply by meftunca** [reply]
I'm trying to develop my own data format and the Beve library seems to solve my problem for a long time :). Will you provide an interface for Beve where we can write plugins for different data formats?

**Reply by stephenberry** [reply]
Thanks for your interest. The BEVE specification is now formalized, but all the different programming language and data format translations are still in development. There is good support in C++ for reading and writing. The [BEVE repository](https://github.com/stephenberry/beve) has a MatLab reader that my company uses and I just added a javascript reader/writer that can be used on the web and with Node.js.

Can you provide a little more explanation of what you mean by "an interface for Beve where we can write plugins for different data formats"? Is there a specific data format that you want to translate to/from? Or, is there a programming language you need support for? What programming language would this plugin be in?

**Reply by stephenberry** [reply]
Glaze also provide a `beve_to_json` direct conversion function: https://github.com/stephenberry/glaze/blob/main/include/glaze/binary/beve_to_json.hpp

**Reply by meftunca** [reply]
What I meant by plugin, I asked to be able to diversify it to support some of the sql data types. It's actually not very important. Collection structure will already contain type information.

**Reply by meftunca** [reply]
I can write the golang port if you want

**Reply by stephenberry** [reply]
Gotcha, more sql utilities would make sense in the future, but you probably want to write them for your custom needs right now.

I can add you as a contributor to the BEVE repository to add golang support there. Unless you'd prefer to create your own repository.

**Reply by meftunca** [reply]
Of course, I would be happy to join the ranks of contributors :)

**Reply by stephenberry** [reply]
Thanks! I sent you an invite to [BEVE](https://github.com/stephenberry/beve) to make you a collaborator

**Reply by meftunca** [reply]
Do we have sample “.beve” files? I have almost completed the integration at alpha level for Golang. After testing, I will share a stable version with additional functions such as “fromJSON”, “toJSON”

**Reply by meftunca** [reply]
In addition to that, I would like to give you a piece of advice. The “Beve” project is a really good project. Why don't you turn it into an organization and create a repo for each language?  Each language repo can develop its own specifications and algorithms. Also, package managers can be used for better quality versioning.

**Reply by stephenberry** [reply]
I'll add example test files for BEVE when I get the chance, this will definitely be helpful. I also think your idea of making an organization for BEVE with separate repos for languages is a good idea. I'll try to do this soon.

**Reply by stephenberry** [reply]
I just moved the repository to the new [beve-org](https://github.com/beve-org). I will be adding more repositories for different language implementations and will give you access. It might just be a slow process.
